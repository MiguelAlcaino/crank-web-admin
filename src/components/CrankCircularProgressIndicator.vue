<script setup lang="ts">
defineProps<{
  text?: string
}>()
</script>

<template>
  <div class="spinner-border crankSpiner" role="status">
    <span class="sr-only">Loading...</span>
  </div>
  <p v-if="text">
    <strong>{{ text }}</strong>
  </p>
</template>

<style scoped>
.crankSpiner {
  color: #ff6f60;
}

span {
  font-family: 'Avenir', sans-serif !important;
}
</style>
