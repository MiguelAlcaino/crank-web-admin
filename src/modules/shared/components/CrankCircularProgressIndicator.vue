<script setup lang="ts">
defineProps<{
  text?: string
}>()
</script>

<template>
  <div class="spinner-border crankSpinner" role="status">
    <span class="sr-only">Loading...</span>
  </div>
  <p v-if="text">
    <strong>{{ text }}</strong>
  </p>
</template>

<style scoped>
.crankSpinner {
  color: #ff6f60;
}

span {
  font-family: 'Avenir', sans-serif !important;
}
</style>
