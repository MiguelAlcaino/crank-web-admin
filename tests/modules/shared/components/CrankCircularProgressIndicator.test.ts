import { mount } from '@vue/test-utils'
import CrankCircularProgressIndicator from '@/modules/shared/components/CrankCircularProgressIndicator.vue'

describe('<CrankCircularProgressIndicator />', () => {
  test('should match snapshot', () => {
    const wrapper = mount(CrankCircularProgressIndicator, {
      props: {}
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render value text correctly', () => {
    const text = 'Please wait'

    const wrapper = mount(CrankCircularProgressIndicator, {
      props: {
        text: text
      }
    })

    expect(wrapper.find('p strong').text()).toBe(text)
  })

  test('renders without text prop', () => {
    const wrapper = mount(CrankCircularProgressIndicator)
    expect(wrapper.find('.crankSpinner').exists()).toBe(true)
    expect(wrapper.find('p').exists()).toBe(false)
  })

  test('renders with text prop', () => {
    const text = 'Please wait'
    const wrapper = mount(CrankCircularProgressIndicator, {
      props: { text }
    })
    expect(wrapper.find('.crankSpinner').exists()).toBe(true)
    expect(wrapper.find('p').exists()).toBe(true)
    expect(wrapper.find('p strong').text()).toBe(text)
  })
})
