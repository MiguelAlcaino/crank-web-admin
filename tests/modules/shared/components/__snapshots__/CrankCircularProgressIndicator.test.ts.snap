// Vitest Snapshot v1

exports[`<CrankCircularProgressIndicator /> > should match snapshot 1`] = `
"<div data-v-4b55f3a8=\\"\\" class=\\"spinner-border crankSpiner\\" role=\\"status\\"><span data-v-4b55f3a8=\\"\\" class=\\"sr-only\\">Loading...</span></div>
<!--v-if-->"
`;
